### PR TITLE
Yaml version support [14068]

### PR DESCRIPTION
**Pull request by jparisu**
support different YAML versions regarding a yaml tag

### CODE REVIEW

File: ddsrouter_yaml/src/cpp/YamlReader.cpp [hunk 1]
```diff
@@ -80,13 +81,27 @@ Yaml YamlReader::get_value_in_tag(
     }
 }
 
+template <>
+YamlReaderVersion YamlReader::get<YamlReaderVersion>(
+        const Yaml& yml,
+        const YamlReaderVersion /* Not Specialization */)
```
Comment (rsanchez15) [inline ddsrouter_yaml/src/cpp/YamlReader.cpp:87]: Remove the `/* Not Specialization */` comment as this is no longer a template. Instead use the following:

```suggestion
        const YamlReaderVersion /* version = LATEST */)
```

**Review by rsanchez15 — CHANGES_REQUESTED** [review]

File: ddsrouter_core/src/cpp/types/dds/GuidPrefix.cpp [hunk 2]
```diff
@@ -44,6 +44,12 @@ GuidPrefix::GuidPrefix (
     // TODO launch exception
 }
 
+GuidPrefix::GuidPrefix (
+        const char* str_prefix)
+    : GuidPrefix(std::string(str_prefix))
```
[the remark this answers is not in the document]
Comment (jparisu) [reply ddsrouter_core/src/cpp/types/dds/GuidPrefix.cpp:49]: It is not a cast, it is a `std::string` constructor

File: ddsrouter_yaml/include/ddsrouter_yaml/YamlReader.hpp [hunk 3]
```diff
@@ -28,74 +28,110 @@ namespace yaml {
...
+     */
+    LATEST,
 };
 
 using TagType = std::string;
 
 /**
  * @brief Class that encapsulates methods to get values from Yaml Node.
  *
+ * TODO: Check if default arguments are needed. They may not be, as version should always be given
```
[the remark this answers is not in the document]
Comment (jparisu) [reply ddsrouter_yaml/include/ddsrouter_yaml/YamlReader.hpp:65]: I removed it

File: ddsrouter_yaml/include/ddsrouter_yaml/yaml_configuration_tags.hpp [hunk 4]
```diff
@@ -31,6 +31,14 @@ namespace yaml {
 //! Retrieve a set with every tag used in the configuration of the DDSRouter or the Participants
 std::set<std::string> ddsrouter_tags() noexcept;
 
+//! Whether a string is a tag / key word
+bool is_tag(const std::string& str) noexcept;
+
+// Yaml related tags
+constexpr const char* VERSION_TAG("version");       //! Version to parse the yaml with
+constexpr const char* VERSION_TAG_V_1_0("v1.0");    //! Version v1.0
+constexpr const char* VERSION_TAG_V_2_0("v2.0");    //! Version v2.0
```
[the remark this answers is not in the document]
Comment (jparisu) [reply ddsrouter_yaml/include/ddsrouter_yaml/yaml_configuration_tags.hpp:40]: We decided to leave it this way to avoid the yaml parse as a number

**Pull request merged**